app/instrumentation-tests/src/org/commcare/androidTests/DateWidgetsTests.kt

```diff
@@ -0,0 +1,207 @@
+package org.commcare.androidTests
+
+
+import android.widget.DatePicker
+import androidx.test.espresso.Espresso.*
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.contrib.PickerActions
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import org.commcare.annotations.BrowserstackTests
+import org.commcare.dalvik.R
+import org.commcare.utils.InstrumentationUtility
+import org.commcare.utils.isPresent
+import org.hamcrest.Matchers
+import org.hamcrest.Matchers.allOf
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.text.SimpleDateFormat
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.*
+import kotlin.math.absoluteValue
+
+
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+@BrowserstackTests
+class DateWidgetsTests: BaseTest() {
+    companion object {
+        const val CCZ_NAME = "date_widgets_tests.ccz"
+        const val APP_NAME = "Date Widgets"
+        val listOfNepaliMonths = listOf("Baishakh","Jestha","Ashadh","Shrawan",
+                            "Bhadra","Ashwin","Kartik","Mangsir","Poush","Magh",
+                            "Falgun","Chaitra")
+        val listOfEthiopianMonths = listOf("Säne","Hämle","Nähäse","P’agume",
+                "Mäskäräm","T’ïk’ïmt","Hïdar","Tahsas","T’ïr","Yäkatit",
+                "Mägabit","Miyaziya")
+
+    }
+
+    @Before
+    fun setup() {
+        installApp(APP_NAME, CCZ_NAME)
+        InstrumentationUtility.login("test1", "123")
+    }
+
+    /**
+     * Adding the teardown method so the tests doesnot fail after only thefirst execution, when executed as a whole.
+     */
+
+    @After
+    fun teardown(){
+        InstrumentationUtility.logout()
+    }
+
+    @Test
+    fun testDateWidgets(){
+        testNepaliDateWidgets()
+        testEthiopianDateWidgets()
+    }
+
+
+    fun testNepaliDateWidgets(){
+
+        InstrumentationUtility.openModule(0)
```

Comment:
since this is the only line that's different between `testNepaliDateWidgets` and `testEthiopianDateWidgets` you can merge the 2 into a single function that takes the module number and list of months as an argument.

Reply:
made the changes